File: backend/internal/model/membership.go
package model

import "time"

type MembershipAudience string
type MembershipBillingCycle string
type MembershipSubscriptionStatus string
type MembershipOrderStatus string
type TeamStatus string
type TeamMemberRole string
type TeamMemberStatus string
type TeamInvitationStatus string
type InvoiceRequestStatus string

const (
	MembershipAudiencePersonal MembershipAudience = "personal"
	MembershipAudienceTeam     MembershipAudience = "team"

	MembershipBillingCycleFree  MembershipBillingCycle = "free"
	MembershipBillingCycleMonth MembershipBillingCycle = "month"
	MembershipBillingCycleYear  MembershipBillingCycle = "year"

	MembershipSubscriptionActive    MembershipSubscriptionStatus = "active"
	MembershipSubscriptionExpired   MembershipSubscriptionStatus = "expired"
	MembershipSubscriptionCancelled MembershipSubscriptionStatus = "cancelled"

	MembershipOrderPending   MembershipOrderStatus = "pending"
	MembershipOrderPaid      MembershipOrderStatus = "paid"
	MembershipOrderCancelled MembershipOrderStatus = "cancelled"
	MembershipOrderRefunded  MembershipOrderStatus = "refunded"

	TeamStatusActive   TeamStatus = "active"
	TeamStatusDisabled TeamStatus = "disabled"

	TeamMemberRoleOwner  TeamMemberRole = "owner"
	TeamMemberRoleAdmin  TeamMemberRole = "admin"
	TeamMemberRoleMember TeamMemberRole = "member"

	TeamMemberStatusActive  TeamMemberStatus = "active"
	TeamMemberStatusRemoved TeamMemberStatus = "removed"

	TeamInvitationStatusPending  TeamInvitationStatus = "pending"
	TeamInvitationStatusAccepted TeamInvitationStatus = "accepted"
	TeamInvitationStatusRevoked  TeamInvitationStatus = "revoked"
	TeamInvitationStatusExpired  TeamInvitationStatus = "expired"

	InvoiceRequestStatusPending  InvoiceRequestStatus = "pending"
	InvoiceRequestStatusIssued   InvoiceRequestStatus = "issued"
	InvoiceRequestStatusRejected InvoiceRequestStatus = "rejected"
)

// MembershipPlan 是可销售权益的唯一配置源，历史订单与订阅通过快照保持审计稳定。
type MembershipPlan struct {
	ID                        string                 `json:"id" gorm:"primaryKey;size:36"`
	Code                      string                 `json:"code" gorm:"uniqueIndex;size:80"`
	Name                      string                 `json:"name" gorm:"size:80"`
	Tier                      string                 `json:"tier" gorm:"index;size:32"`
	Audience                  MembershipAudience     `json:"audience" gorm:"index;size:24"`
	BillingCycle              MembershipBillingCycle `json:"billingCycle" gorm:"index;size:24"`
	PriceCents                int64                  `json:"priceCents"`
	OriginalPriceCents        int64                  `json:"originalPriceCents"`
	Currency                  string                 `json:"currency" gorm:"size:12"`
	CreditsPerPeriod          int64                  `json:"creditsPerPeriod"`
	ImageConcurrency          int                    `json:"imageConcurrency"`
	VideoConcurrency          int                    `json:"videoConcurrency"`
	UnlimitedTaskQueue        bool                   `json:"unlimitedTaskQueue"`
	TeamStorageBytes          int64                  `json:"teamStorageBytes"`
	SharedAssetsEnabled       bool                   `json:"sharedAssetsEnabled"`
	ProjectPermissionsEnabled bool                   `json:"projectPermissionsEnabled"`
	InvoicingEnabled          bool                   `json:"invoicingEnabled"`
	CommercialUseEnabled      bool                   `json:"commercialUseEnabled"`
	TopupDiscountBasisPoints  int                    `json:"topupDiscountBasisPoints"`
	MinSeats                  int                    `json:"minSeats"`
	MaxSeats                  int                    `json:"maxSeats"`
	BenefitsJSON              string                 `json:"benefitsJson" gorm:"type:text"`
	Enabled                   bool                   `json:"enabled" gorm:"index"`
	SortOrder                 int                    `json:"sortOrder" gorm:"index"`
	CreatedAt                 time.Time              `json:"createdAt"`
	UpdatedAt                 time.Time              `json:"updatedAt"`
}

type Team struct {
	ID          string     `json:"id" gorm:"primaryKey;size:36"`
	OwnerUserID string     `json:"ownerUserId" gorm:"index;size:36"`
	Name        string     `json:"name" gorm:"size:120"`
	Status      TeamStatus `json:"status" gorm:"index;size:24"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
}

type TeamMember struct {
	ID                             string           `json:"id" gorm:"primaryKey;size:36"`
	TeamID                         string           `json:"teamId" gorm:"uniqueIndex:idx_team_member,priority:1;size:36"`
	UserID                         string           `json:"userId" gorm:"uniqueIndex:idx_team_member,priority:2;index;size:36"`
	Role                           TeamMemberRole   `json:"role" gorm:"size:24"`
	Status                         TeamMemberStatus `json:"status" gorm:"index;size:24"`
	MonthlyCreditLimitMicrocredits int64            `json:"monthlyCreditLimitMicrocredits"`
	CreatedAt                      time.Time        `json:"createdAt"`
	UpdatedAt                      time.Time        `json:"updatedAt"`
}

// TeamInvitation 只持久化邀请凭证哈希，原始凭证仅在创建成功时返回一次。
type TeamInvitation struct {
	ID               string               `json:"id" gorm:"primaryKey;size:36"`
	TeamID           string               `json:"teamId" gorm:"uniqueIndex:idx_team_invitation_email,priority:1;index;size:36"`
	InviterUserID    string               `json:"inviterUserId" gorm:"index;size:36"`
	Email            string               `json:"email" gorm:"uniqueIndex:idx_team_invitation_email,priority:2;size:160"`
	Role             TeamMemberRole       `json:"role" gorm:"size:24"`
	Status           TeamInvitationStatus `json:"status" gorm:"index;size:24"`
	TokenHash        string               `json:"-" gorm:"uniqueIndex;size:64"`
	ExpiresAt        time.Time            `json:"expiresAt" gorm:"index"`
	AcceptedByUserID string               `json:"acceptedByUserId,omitempty" gorm:"index;size:36"`
	AcceptedAt       *time.Time           `json:"acceptedAt,omitempty"`
	RevokedAt        *time.Time           `json:"revokedAt,omitempty"`
	CreatedAt        time.Time            `json:"createdAt"`
	UpdatedAt        time.Time            `json:"updatedAt"`
}

// TeamAuditEvent 是团队写操作的追加式审计记录，不允许由普通更新覆盖。
type TeamAuditEvent struct {
	ID                 string    `json:"id" gorm:"primaryKey;size:36"`
	TeamID             string    `json:"teamId" gorm:"index;size:36"`
	ActorUserID        string    `json:"actorUserId" gorm:"index;size:36"`
	Action             string    `json:"action" gorm:"index;size:64"`
	TargetUserID       string    `json:"targetUserId,omitempty" gorm:"index;size:36"`
	TargetInvitationID string    `json:"targetInvitationId,omitempty" gorm:"index;size:36"`
	MetadataJSON       string    `json:"metadataJson" gorm:"type:text"`
	CreatedAt          time.Time `json:"createdAt" gorm:"index"`
}

// TeamCreditAccount 是团队积分的唯一余额真源；成员调用只记录 actor，不落入个人积分账户。
type TeamCreditAccount struct {
	TeamID                string    `json:"teamId" gorm:"primaryKey;size:36"`
	AvailableMicrocredits int64     `json:"availableMicrocredits"`
	ReservedMicrocredits  int64     `json:"reservedMicrocredits"`
	Version               int64     `json:"version"`
	CreatedAt             time.Time `json:"createdAt"`
	UpdatedAt             time.Time `json:"updatedAt"`
}

// TeamCreditLedgerEntry 是追加式团队积分流水，ActorUserID 用于成员额度与审计归因。
type TeamCreditLedgerEntry struct {
	ID                         string           `json:"id" gorm:"primaryKey;size:36"`
	TeamID                     string           `json:"teamId" gorm:"index:idx_team_credit_ledger_created,priority:1;size:36"`
	ActorUserID                string           `json:"actorUserId" gorm:"index:idx_team_credit_actor_created,priority:1;size:36"`
	Type                       CreditLedgerType `json:"type" gorm:"size:32;index"`
	AmountMicrocredits         int64            `json:"amountMicrocredits"`
	AvailableDeltaMicrocredits int64            `json:"availableDeltaMicrocredits"`
	ReservedDeltaMicrocredits  int64            `json:"reservedDeltaMicrocredits"`
	AvailableAfterMicrocredits int64            `json:"availableAfterMicrocredits"`
	ReservedAfterMicrocredits  int64            `json:"reservedAfterMicrocredits"`
	BillingOrderID             string           `json:"billingOrderId,omitempty" gorm:"index;size:36"`
	Model                      string           `json:"model,omitempty" gorm:"size:120;index"`
	ChannelID                  string           `json:"channelId,omitempty" gorm:"size:36;index"`
	Scene                      string           `json:"scene,omitempty" gorm:"size:80;index"`
	Note                       string           `json:"note,omitempty" gorm:"size:500"`
	ReferenceKey               *string          `json:"referenceKey,omitempty" gorm:"size:180;uniqueIndex"`
	CreatedAt                  time.Time        `json:"createdAt" gorm:"index:idx_team_credit_ledger_created,priority:2;index:idx_team_credit_actor_created,priority:2"`
}

// InvoiceRequest 只表示开票业务状态；真实电子发票必须由后台或已配置的税务服务商回填。
type InvoiceRequest struct {
	ID                string               `json:"id" gorm:"primaryKey;size:36"`
	UserID            string               `json:"userId" gorm:"index;size:36"`
	TeamID            string               `json:"teamId,omitempty" gorm:"index;size:36"`
	MembershipOrderID string               `json:"membershipOrderId" gorm:"uniqueIndex;size:36"`
	Title             string               `json:"title" gorm:"size:200"`
	TaxNumber         string               `json:"taxNumber,omitempty" gorm:"size:80"`
	Email             string               `json:"email" gorm:"size:160"`
	AmountCents       int64                `json:"amountCents"`
	Status            InvoiceRequestStatus `json:"status" gorm:"index;size:24"`
	InvoiceNumber     string               `json:"invoiceNumber,omitempty" gorm:"size:120"`
	InvoiceURL        string               `json:"invoiceUrl,omitempty"`
	ResolutionNote    string               `json:"resolutionNote,omitempty" gorm:"size:500"`
	ResolvedBy        string               `json:"resolvedBy,omitempty" gorm:"index;size:36"`
	ResolvedAt        *time.Time           `json:"resolvedAt,omitempty"`
	CreatedAt         time.Time            `json:"createdAt" gorm:"index"`
	UpdatedAt         time.Time            `json:"updatedAt"`
}

type MembershipOrder struct {
	ID               string                `json:"id" gorm:"primaryKey;size:36"`
	OrderNumber      string                `json:"orderNumber" gorm:"uniqueIndex;size:40"`
	UserID           string                `json:"userId" gorm:"index;size:36"`
	IdempotencyKey   string                `json:"-" gorm:"size:120;not null;default:''"`
	RequestHash      string                `json:"-" gorm:"size:64;not null;default:''"`
	TeamID           string                `json:"teamId,omitempty" gorm:"index;size:36"`
	PlanID           string                `json:"planId" gorm:"index;size:36"`
	Seats            int                   `json:"seats"`
	UnitPriceCents   int64                 `json:"unitPriceCents"`
	TotalPriceCents  int64                 `json:"totalPriceCents"`
	Currency         string                `json:"currency" gorm:"size:12"`
	Status           MembershipOrderStatus `json:"status" gorm:"index;size:24"`
	PaymentProvider  string                `json:"paymentProvider" gorm:"size:40"`
	ProviderTradeNo  string                `json:"providerTradeNo" gorm:"size:120"`
	PlanSnapshotJSON string                `json:"planSnapshotJson" gorm:"type:text"`
	ResolvedBy       string                `json:"resolvedBy,omitempty" gorm:"size:36"`
	ResolutionNote   string                `json:"resolutionNote,omitempty" gorm:"size:500"`
	PaidAt           *time.Time            `json:"paidAt,omitempty"`
	CreatedAt        time.Time             `json:"createdAt"`
	UpdatedAt        time.Time             `json:"updatedAt"`
}

type MembershipSubscription struct {
	ID               string                       `json:"id" gorm:"primaryKey;size:36"`
	UserID           string                       `json:"userId" gorm:"index;size:36"`
	TeamID           string                       `json:"teamId,omitempty" gorm:"index;size:36"`
	PlanID           string                       `json:"planId" gorm:"index;size:36"`
	OrderID          string                       `json:"orderId,omitempty" gorm:"uniqueIndex;size:36"`
	Status           MembershipSubscriptionStatus `json:"status" gorm:"index;size:24"`
	Seats            int                          `json:"seats"`
	PlanSnapshotJSON string                       `json:"planSnapshotJson" gorm:"type:text"`
	StartsAt         time.Time                    `json:"startsAt" gorm:"index"`
	EndsAt           *time.Time                   `json:"endsAt,omitempty" gorm:"index"`
	CreatedAt        time.Time                    `json:"createdAt"`
	UpdatedAt        time.Time                    `json:"updatedAt"`
}

File: web/src/services/api/membership.ts
import axios from "axios";

const api = axios.create({ baseURL: import.meta.env.VITE_CANVAS_BACKEND_URL || "/api", withCredentials: true });

type BackendEnvelope<T> = { code: number; data: T; msg: string };

async function request<T>(promise: Promise<{ data: BackendEnvelope<T> }>): Promise<T> {
    try {
        const response = await promise;
        if (response.data.code !== 0) throw new Error(response.data.msg || "请求失败");
        return response.data.data;
    } catch (error) {
        if (axios.isAxiosError<BackendEnvelope<unknown>>(error)) throw new Error(error.response?.data?.msg || error.message || "请求失败");
        throw error;
    }
}

export type MembershipAudience = "personal" | "team";
export type MembershipBillingCycle = "free" | "month" | "year";

export type MembershipPlan = {
    id: string;
    code: string;
    name: string;
    tier: string;
    audience: MembershipAudience;
    billingCycle: MembershipBillingCycle;
    priceCents: number;
    originalPriceCents: number;
    currency: string;
    creditsPerPeriod: number;
    imageConcurrency: number;
    videoConcurrency: number;
    unlimitedTaskQueue: boolean;
    teamStorageBytes: number;
    sharedAssetsEnabled: boolean;
    projectPermissionsEnabled: boolean;
    invoicingEnabled: boolean;
    commercialUseEnabled: boolean;
    topupDiscountBasisPoints: number;
    minSeats: number;
    maxSeats: number;
    benefitsJson: string;
    benefits: string[];
    enabled: boolean;
    sortOrder: number;
    createdAt: string;
    updatedAt: string;
};

export type MembershipEntitlement = {
    planId: string;
    planName: string;
    tier: string;
    audience: MembershipAudience;
    isActiveMember: boolean;
    imageConcurrency: number;
    videoConcurrency: number;
    topupDiscountBasisPoints: number;
    unlimitedTaskQueue: boolean;
    teamStorageBytes: number;
    sharedAssetsEnabled: boolean;
    projectPermissionsEnabled: boolean;
    invoicingEnabled: boolean;
    commercialUseEnabled: boolean;
    teamId?: string;
    expiresAt?: string;
};

export type MembershipOrder = {
    id: string;
    orderNumber: string;
    userId: string;
    teamId?: string;
    planId: string;
    seats: number;
    unitPriceCents: number;
    totalPriceCents: number;
    currency: string;
    status: "pending" | "paid" | "cancelled" | "refunded";
    planSnapshotJson: string;
    paymentProvider: string;
    providerTradeNo: string;
    resolutionNote?: string;
    paidAt?: string;
    createdAt: string;
    updatedAt: string;
};

export type Team = { id: string; ownerUserId: string; name: string; status: "active" | "disabled"; createdAt: string; updatedAt: string };

export type MembershipOverview = { entitlement: MembershipEntitlement; orders: MembershipOrder[]; teams: Team[] };

export type MembershipOrderRequestInput = { planId: string; teamId?: string; seats?: number };

export type MembershipOrderRequestIdentity = {
    fingerprint: string;
    key: string;
};

type MembershipOrderSubmissionDependencies = {
    createTeam: (name: string, idempotencyKey: string) => Promise<Team>;
    createOrder: (input: MembershipOrderRequestInput, idempotencyKey: string) => Promise<MembershipOrder>;
    persistResolvedTeamID: (teamID: string) => void;
    persistIdentity: (identity: MembershipOrderRequestIdentity) => void;
};

function normalizeMembershipOrderRequest(input: MembershipOrderRequestInput): Required<MembershipOrderRequestInput> {
    const planId = input.planId.trim();
    const teamId = (input.teamId ?? "").trim();
    const seats = input.seats ?? 1;
    if (!planId) throw new Error("会员套餐 ID 不能为空");
    if (!Number.isSafeInteger(seats) || seats < 1) throw new Error("会员席位数必须是正整数");
    return { planId, teamId, seats };
}

export function bindMembershipOrderRequestIdentity(input: MembershipOrderRequestInput, current: MembershipOrderRequestIdentity | null, nextKey: string): MembershipOrderRequestIdentity {
    const fingerprint = JSON.stringify(normalizeMembershipOrderRequest(input));
    if (current?.fingerprint === fingerprint) return current;
    if (new TextEncoder().encode(nextKey).byteLength < 1 || new TextEncoder().encode(nextKey).byteLength > 120) {
        throw new Error("会员订单幂等 key 必须为 1 到 120 字节");
    }
    return { fingerprint, key: nextKey };
}

export async function submitMembershipOrderRequest(
    input: MembershipOrderRequestInput,
    teamName: string,
    requiresTeam: boolean,
    currentIdentity: MembershipOrderRequestIdentity | null,
    nextKey: string,
    dependencies: MembershipOrderSubmissionDependencies,
): Promise<MembershipOrder> {
    let resolvedInput = normalizeMembershipOrderRequest(input);
    if (!requiresTeam) resolvedInput = { ...resolvedInput, teamId: "" };
    if (requiresTeam && !resolvedInput.teamId) {
        const normalizedTeamName = teamName.trim();
        if (!normalizedTeamName) throw new Error("请输入团队名称");
        const team = await dependencies.createTeam(normalizedTeamName, `team-create:${nextKey}`);
        const resolvedTeamID = team.id.trim();
        if (!resolvedTeamID) throw new Error("新团队创建成功但返回的团队 ID 为空");
        dependencies.persistResolvedTeamID(resolvedTeamID);
        resolvedInput = { ...resolvedInput, teamId: resolvedTeamID };
    }
    const identity = bindMembershipOrderRequestIdentity(resolvedInput, currentIdentity, nextKey);
    dependencies.persistIdentity(identity);
    return dependencies.createOrder(resolvedInput, identity.key);
}

export type MembershipStorefrontPromotion = {
    enabled: boolean;
    title: string;
    subtitle: string;
    subtitleHighlight: string;
    endsAt: string;
};

export type MembershipStorefrontActivity = { icon: string; text: string };

export type MembershipStorefrontCopy = {
    creatorTab: string;
    teamTab: string;
    yearCycle: string;
    monthCycle: string;
    creditStore: string;
    activityHeading: string;
    exclusiveHeading: string;
    generationHeading: string;
    faqHeading: string;
};

export type MembershipStorefrontPlanHighlight = {
    tier: string;
    images: string;
    videos: string;
};

export type MembershipStorefrontGenerationColumn = { key: string; label: string };

export type MembershipStorefrontGenerationRow = {
    model: string;
    icon: string;
    unit: string;
    values: string[];
};

export type MembershipStorefrontGenerationSection = {
    title: string;
    rows: MembershipStorefrontGenerationRow[];
};

export type MembershipStorefrontFAQ = { question: string; answer: string };

export type MembershipStorefrontSetting = {
    promotion: MembershipStorefrontPromotion;
    copy: MembershipStorefrontCopy;
    activities: MembershipStorefrontActivity[];
    commonFeatures: string[];
    exclusiveFeatures: string[];
    planHighlights: MembershipStorefrontPlanHighlight[];
    generationColumns: MembershipStorefrontGenerationColumn[];
    generationSections: MembershipStorefrontGenerationSection[];
    generationFootnote: string;
    membershipNotes: string[];
    faqs: MembershipStorefrontFAQ[];
};

export type MembershipStorefront = {
    presentation: MembershipStorefrontSetting;
    plans: MembershipPlan[];
    serverNow: string;
    updatedAt?: string;
};

export function getMembershipStorefront() {
    return request<MembershipStorefront>(api.get("/membership/storefront"));
}

export function listMembershipPlans() {
    return request<MembershipPlan[]>(api.get("/membership/plans"));
}

export function getMyMembership() {
    return request<MembershipOverview>(api.get("/membership"));
}

export function membershipOrderRequest(input: MembershipOrderRequestInput, idempotencyKey: string) {
    return {
        data: input,
        headers: { "Idempotency-Key": idempotencyKey },
        method: "post" as const,
        url: "/membership/orders",
    };
}

export function createMembershipOrder(input: MembershipOrderRequestInput, idempotencyKey: string) {
    return request<MembershipOrder>(api.request(membershipOrderRequest(input, idempotencyKey)));
}

export function cancelMembershipOrder(id: string) {
    return request<MembershipOrder>(api.post(`/membership/orders/${encodeURIComponent(id)}/cancel`, {}));
}

export function createTeam(name: string, idempotencyKey: string) {
    return request<Team>(api.post("/teams", { name }, { headers: { "Idempotency-Key": idempotencyKey } }));
}

export function listAdminMembershipPlans() {
    return request<MembershipPlan[]>(api.get("/admin/membership/plans"));
}

export function getAdminMembershipStorefront() {
    return request<MembershipStorefront>(api.get("/admin/membership/storefront"));
}

export function updateAdminMembershipStorefront(input: MembershipStorefrontSetting) {
    return request<MembershipStorefront>(api.put("/admin/membership/storefront", input));
}

export type UpdateMembershipPlanInput = {
    name: string;
    priceCents: number;
    originalPriceCents: number;
    creditsPerPeriod: number;
    imageConcurrency: number;
    videoConcurrency: number;
    topupDiscountBasisPoints: number;
    unlimitedTaskQueue: boolean;
    teamStorageBytes: number;
    sharedAssetsEnabled: boolean;
    projectPermissionsEnabled: boolean;
    invoicingEnabled: boolean;
    commercialUseEnabled: boolean;
    minSeats: number;
    maxSeats: number;
    benefits: string[];
    enabled: boolean;
    sortOrder: number;
};

export type InvoiceRequestStatus = "pending" | "issued" | "rejected";

export type InvoiceRequest = {
    id: string;
    userId: string;
    teamId?: string;
    membershipOrderId: string;
    title: string;
    taxNumber?: string;
    email: string;
    amountCents: number;
    status: InvoiceRequestStatus;
    invoiceNumber?: string;
    invoiceUrl?: string;
    resolutionNote?: string;
    resolvedBy?: string;
    resolvedAt?: string;
    createdAt: string;
    updatedAt: string;
};

export function updateAdminMembershipPlan(id: string, input: UpdateMembershipPlanInput) {
    return request<MembershipPlan>(api.patch(`/admin/membership/plans/${encodeURIComponent(id)}`, input));
}

export function listAdminMembershipOrders(page = 1, limit = 30) {
    return request<{ items: MembershipOrder[]; total: number; page: number; limit: number }>(api.get("/admin/membership/orders", { params: { page, limit } }));
}

export function closeAdminMembershipOrder(id: string, input: { note: string }) {
    return request<MembershipOrder>(api.post(`/admin/membership/orders/${encodeURIComponent(id)}/close`, input));
}

export function listMyInvoiceRequests() {
    return request<{ items: InvoiceRequest[] }>(api.get("/membership/invoices"));
}

export function createInvoiceRequest(input: { membershipOrderId: string; title: string; taxNumber?: string; email: string }) {
    return request<InvoiceRequest>(api.post("/membership/invoices", input));
}

export function listAdminInvoiceRequests(status?: InvoiceRequestStatus, page = 1, limit = 30) {
    return request<{ items: InvoiceRequest[]; total: number; page: number; limit: number }>(api.get("/admin/membership/invoices", { params: { status, page, limit } }));
}

export function resolveAdminInvoiceRequest(
    id: string,
    input: {
        status: Exclude<InvoiceRequestStatus, "pending">;
        invoiceNumber?: string;
        invoiceUrl?: string;
        note: string;
    },
) {
    return request<{ resolved: boolean }>(api.post(`/admin/membership/invoices/${encodeURIComponent(id)}/resolve`, input));
}
